src/oscillator.js

```diff
@@ -1,634 +1,630 @@
+ *  shape ('sine'). Additional types include 'triangle',
+ *  'sawtooth' and 'square'. The frequency defaults to
+ *  440 oscillations per second (440Hz, equal to the pitch of an
+ *  'A' note).</p>
+ *
+ *  <p>Set the type of oscillation with setType(), or by instantiating a
+ *  specific oscillator: <a href="/reference/#/p5.SinOsc">p5.SinOsc</a>, <a
+ *  href="/reference/#/p5.TriOsc">p5.TriOsc</a>, <a
+ *  href="/reference/#/p5.SqrOsc">p5.SqrOsc</a>, or <a
+ *  href="/reference/#/p5.SawOsc">p5.SawOsc</a>.
+ *  </p>
+ *
+ *  @class p5.Oscillator
+ *  @constructor
+ *  @param {Number} [freq] frequency defaults to 440Hz
+ *  @param {String} [type] type of oscillator. Options:
+ *                         'sine' (default), 'triangle',
+ *                         'sawtooth', 'square'
+ *  @example
+ *  <div><code>
+ *  let osc, playing, freq, amp;
+ *
+ *  function setup() {
+ *    let cnv = createCanvas(100, 100);
+ *    cnv.mousePressed(playOscillator);
+ *    osc = new p5.Oscillator('sine');
+ *  }
+ *
+ *  function draw() {
+ *    background(220)
+ *    freq = constrain(map(mouseX, 0, width, 100, 500), 100, 500);
+ *    amp = constrain(map(mouseY, height, 0, 0, 1), 0, 1);
+ *
+ *    text('tap to play', 20, 20);
+ *    text('freq: ' + freq, 20, 40);
+ *    text('amp: ' + amp, 20, 60);
+ *
+ *    if (playing) {
+ *      // smooth the transitions by 0.1 seconds
+ *      osc.freq(freq, 0.1);
+ *      osc.amp(amp, 0.1);
+ *    }
+ *  }
+ *
+ *  function playOscillator() {
+ *    // starting an oscillator on a user gesture will enable audio
+ *    // in browsers that have a strict autoplay policy.
+ *    // See also: userStartAudio();
+ *    osc.start();
+ *    playing = true;
+ *  }
+ *
+ *  function mouseReleased() {
+ *    // ramp amplitude to 0 over 0.5 seconds
+ *    osc.amp(0, 0.5);
+ *    playing = false;
+ *  }
+ *  </code> </div>
+ */
+p5.Oscillator = function (freq, type) {
+  if (typeof freq === 'string') {
+    let f = type;
+    type = freq;
+    freq = f;
+  }
+  if (typeof type === 'number') {
+    let f = type;
+    type = freq;
+    freq = f;
+  }
+  this.started = false;
+
+  // components
+  this.phaseAmount = undefined;
+  this.oscillator = p5sound.audiocontext.createOscillator();
+  this.f = freq || 440.0; // frequency
+  this.oscillator.type = type || 'sine';
+  this.oscillator.frequency.setValueAtTime(
+    this.f,
+    p5sound.audiocontext.currentTime
+  );
+
+  // connections
+  this.output = p5sound.audiocontext.createGain();
+
+  this._freqMods = []; // modulators connected to this oscillator's frequency
+
+  // set default output gain to 0.5
+  this.output.gain.value = 0.5;
+  this.output.gain.setValueAtTime(0.5, p5sound.audiocontext.currentTime);
+
+  this.oscillator.connect(this.output);
+  // stereo panning
+  this.panPosition = 0.0;
+  this.connection = p5sound.input; // connect to p5sound by default
+  this.panner = new p5.Panner(this.output, this.connection, 1);
```

Comment:
just an observation: there are a few examples like this where we're dependent on the order of imports. If app.js imported oscillator before panner, `p5.Panner` would not exist.

A better approach would be to `import p5Panner from 'panner'` here in the oscillator module.

This doesn't need to be done as part of this PR, but I noticed we're not exporting anything from p5.Panner right now...should we be consistent and have at least a default export for every module?

Reply:
we ll see if we need to export smth out of  them in future !